### PR TITLE
modemmanager: add openrc init script

### PR DESCRIPTION
**Pull request by mtudan**
Add modemmanager bbappend file with openrc init file.

Signed-off-by: Matija Tudan [tudan.matija@gmail.com](mailto:tudan.matija@gmail.com)

### CODE REVIEW

**Comment by mtudan** [comment]
@jsbronder forked from your `next` branch.

File: conf/layer.conf [hunk 1]
```diff
@@ -19,6 +19,7 @@ BBFILE_PATTERN_openrc-layer := "^${LAYERDIR}/"
 
 BBFILES_DYNAMIC += " \
     networking-layer:${LAYERDIR}/dynamic-layers/meta-networking/*/*.bbappend \
+    oe-layer:${LAYERDIR}/dynamic-layers/meta-oe/*/*.bbappend \
```
Comment (jsbronder) [inline conf/layer.conf:22]: This needs to be `openembedded-layer`, not `oe-layer`.  You can tell from what the layer adds to `BBFILE_COLLECTIONS` (http://cgit.openembedded.org/meta-openembedded/tree/meta-oe/conf/layer.conf#n17).

For testing that it's getting correctly applied, you can do something like `bitbake -f -c package modemmanager` and then look in your `TMPDIR` (`cat tmp/build-glibc/work/core2-64-oe-linux/modemmanager/1.18.10-r0/packages-split/modemmanager/etc/init.d/modemmanager` for me).

File: dynamic-layers/meta-oe/modemmanager/files/modemmanager.initd [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/sbin/openrc-run
+
+supervisor=supervise-daemon
+command=/usr/sbin/ModemManager
+command_args="${modemmanager_opts}"
```
Comment (jsbronder) [inline dynamic-layers/meta-oe/modemmanager/files/modemmanager.initd:5]: `modemmanager_opts` isn't defined anywhere.  To follow what a lot of the other init scripts are doing in this layer, simply add a `modemmanager.confd` that has a default `COMMAND_ARGS` (https://github.com/jsbronder/meta-openrc/blob/master/recipes-appends/oe-core/acpid/acpid.confd#L2) which `command_args` is set to here along with any mandantory arguments (https://github.com/jsbronder/meta-openrc/blob/master/recipes-appends/oe-core/acpid/acpid.initd#L6).

File: conf/layer.conf (hunk 1 above)
Comment (jsbronder) [reply conf/layer.conf:22]: Though, honestly, for testing you should really be starting up an image with these scripts and running them to make sure they're working.

Comment (mtudan) [reply conf/layer.conf:22]: > This needs to be `openembedded-layer`, not `oe-layer`. You can tell from what the layer adds to `BBFILE_COLLECTIONS` (http://cgit.openembedded.org/meta-openembedded/tree/meta-oe/conf/layer.conf#n17).
> 
> For testing that it's getting correctly applied, you can do something like `bitbake -f -c package modemmanager` and then look in your `TMPDIR` (`cat tmp/build-glibc/work/core2-64-oe-linux/modemmanager/1.18.10-r0/packages-split/modemmanager/etc/init.d/modemmanager` for me).

There it is! Couldn't find that in the moment of preparing this, thanks!

**Pull request merged**